# Solvency Ratios

Previously, we discussed [Liquidity Ratios](https://www.linkedin.com/pulse/liquidity-ratios-jon-placa) and their use as indicators of a company's short-term financial stability. Carrying the theme of "financial stability" forward, we will investigate [Solvency Ratios](http://www.investopedia.com/terms/s/solvency.asp), a company's ability to meet *long-term* obligations (including interest payments).

Similar to our [Liquidity Ratio example](https://www.linkedin.com/pulse/liquidity-ratios-jon-placa), we will be using [Quandl's API](quandl.com) to pull Boeing's ([BA](http://financials.morningstar.com/ratios/r.html?t=BA)) financials.


In [1]:
#Importing Quandl API
import quandl

#API Key Application
quandl.ApiConfig.api_key = '[API KEY]'
quandl.ApiConfig.api_version = '2015-04-09'

#Pull in Quandl data for BA (Boeing Co.)
data = quandl.get_table('ZACKS/FC', ticker='BA', per_type='A')
data.head(10)

m_ticker ticker  comp_name           comp_name_2 exchange currency_code  \
None                                                                           
0          BA     BA  BOEING CO  Boeing Company (The)     NYSE           USD   
1          BA     BA  BOEING CO  Boeing Company (The)     NYSE           USD   
2          BA     BA  BOEING CO  Boeing Company (The)     NYSE           USD   
3          BA     BA  BOEING CO  Boeing Company (The)     NYSE           USD   
4          BA     BA  BOEING CO  Boeing Company (The)     NYSE           USD   
5          BA     BA  BOEING CO  Boeing Company (The)     NYSE           USD   

     per_end_date per_type per_code  per_fisc_year       ...         \
None                                                     ...          
0      2011-12-31        A     None           2011       ...          
1      2012-12-31        A     None           2012       ...          
2      2013-12-31        A     None           2013       ...          
3      2014-12-31        A     None           2014       ...          
4      2015-12-31        A     None           2015       ...          
5      2016-12-31        A     None           2016       ...          

      stock_based_compsn_qd  cash_flow_oper_activity_qd  \
None                                                      
0                      None                        None   
1                      None                        None   
2                      None                        None   
3                      None                        None   
4                      None                        None   
5                      None                        None   

      net_change_prop_plant_equip_qd comm_stock_div_paid_qd  \
None                                                          
0                               None                   None   
1                               None                   None   
2                               None                   None   
3                               None                   None   
4                               None                   None   
5                               None                   None   

     pref_stock_div_paid_qd  tot_comm_pref_stock_div_qd  wavg_shares_out  \
None                                                                       
0                      None                        None            744.1   
1                      None                        None            755.7   
2                      None                        None            758.9   
3                      None                        None            727.6   
4                      None                        None            686.9   
5                      None                        None            635.5   

      wavg_shares_out_diluted  eps_basic_net  eps_diluted_net  
None                                                           
0                       750.6           5.39             5.34  
1                       761.5           5.15             5.11  
2                       767.6           6.03             5.96  
3                       736.7           7.47             7.38  
4                       695.0           7.52             7.44  
5                       642.8           7.70             7.61  

[6 rows x 249 columns]

## Debt-to-Equity Ratio

A company's Debt-to-Equity Ratio can be a good indicator of the inherent risk associated with a company. The Debt-to-Equity rato indicates how much debt a company has accrued versus shareholder equity in order to finance it's total assets. The Debt-to-Equity Ratio is as follows:

**Debt-to-Equity Ratio = (Short-term Debt + Long-term Debt) / Shareholder Equity**

In [3]:
shorttermDebt = data[data['per_fisc_year']==2016]['curr_portion_debt']
print (shorttermDebt)
longtermDebt = data[data['per_fisc_year']==2016]['tot_lterm_debt']
print (longtermDebt)

shareholderEquity = data[data['per_fisc_year']==2016]['tot_share_holder_equity']


debtToEquity = (shorttermDebt + longtermDebt) / shareholderEquity
debtToEquity


None
5    384.0
Name: curr_portion_debt, dtype: float64
None
5    9568.0
Name: tot_lterm_debt, dtype: float64


None
5    11.347777
dtype: float64

When we compare Boeing's Debt-to-Equity ratio of 11.3 to the [industry average](https://www.stock-analysis-on.net/NYSE/Company/Boeing-Co/Ratios/Long-term-Debt-and-Solvency), we can see a *massive* spike in Boeing's debt-to-equity ratio. After analyzing [Boeing's Debt and Shareholder equity levels of the previous years](http://financials.morningstar.com/ratios/r.html?t=BA&region=usa&culture=en-US), we can see that shareholder equity was greatly reduced. Further investigation leads us to news that Boeing initiatied a [Stock repurchase program in 2016](http://investors.boeing.com/investors/investor-news/press-release-details/2016/Boeing-Board-Raises-Dividend-Renews-Share-Repurchase-Authorization/default.aspx) - effectively reducing Shareholders Equity. 

The share buybacks are denoted in changes to [Treasury Stocks](http://www.investinganswers.com/financial-dictionary/stock-market/treasury-stock-2519) on the Balance Sheet. Below we can see how Treasury Stock changes by ~$7 Billion from 2015 to 2016 - indicating a repurchase program that is consistent with [Boeing's repurchase announcement](http://investors.boeing.com/investors/investor-news/press-release-details/2016/Boeing-Board-Raises-Dividend-Renews-Share-Repurchase-Authorization/default.aspx).


In [11]:
treasuryStock2015 = data[data['per_fisc_year']==2015]['treas_stock']
print (treasuryStock2015)
treasuryStock2016 = data[data['per_fisc_year']==2016]['treas_stock']
print (treasuryStock2016)


None
4    29568.0
Name: treas_stock, dtype: float64
None
5    36097.0
Name: treas_stock, dtype: float64


*NOTE: Companies generally will repurchase shares if they believe their stock is currently undervalued. By removing shares from the marketplace, the remaining shares will typically increase in value (since the denominator in "Earnings-per-share" is reduced, thus inflating the price of the stock). Furthermore, with less shares on the market, greater dividends will be paid out to the remaining shareholders.*

## Debt-to-Assets Ratio

The Debt-to-Asset Ratio indicates the percentage of assets that are funded through debt obligations. Ideally, the Debt-to-Asset Ratio should be low; a low Debt-to-Asset ratio showcases that the company is able to finance the majority of it's assets through Revenue and/or Shareholder Equity purchases. 

The Debt-to-Assets Ratio is as follows:

**Debt-to-Assets = (Short-term Debt + Long-term Debt) / Total Assets**

In [4]:
totalAssets = data[data['per_fisc_year']==2016]['tot_asset']

debtToAsset = (shorttermDebt + longtermDebt) / totalAssets
debtToAsset

None
5    0.110581
dtype: float64

We can see the Boeing is *not* reliant on Debt to finance the majority of it's Assets; only 11% of assets are funded through debt. Therefore, Boeing despite their debt:equity ratio (note the stock buyback program), is still a stable company. 

## Interest Coverage Ratio

With debt, comes interest. As a result, the Interest Coverage Ratio is a worthy solvency indicator that should be applied to Boeing's financials. The Interest Coverage Ratio will help analysts determine whether or not a company can meet interest expense payments with its Operating Income [(EBIT)](http://www.investopedia.com/terms/e/ebit.asp).

The Interest Coverage Ratio is as following:

**Interest Coverage Ratio = EBIT / Interest Expense**

In [20]:
ebit = data[data['per_fisc_year']==2016]['ebit']

interestExpense = data[data['per_fisc_year']==2016]['non_oper_int_exp']

interestCoverageRatio = ebit / interestExpense
interestCoverageRatio

None
5    19.25817
dtype: float64

Our analysis indicates that Boeing can easily cover any interest expenses that it incurs on an annual basis. Further investigation yields that Boeing is borrowing its debt at a favorable, 3% interest rate.

In [21]:
shorttermDebt = data[data['per_fisc_year']==2016]['curr_portion_debt']
longtermDebt = data[data['per_fisc_year']==2016]['tot_lterm_debt']

interestRate = ((interestExpense) / (shorttermDebt + longtermDebt)) * 100
interestRate

None
5    3.074759
dtype: float64

## Conclusion

Solvency Ratios are critical in determining the long-term stability of a company - when analyzing a company's solvency, analysts must be aware of longer-term events such as stock repurchases and/or upcoming interest obligations. Furthermore, Solvency Ratios must always be compared to industry averages to determine whether or not the associated company is an outlier.